src/components/RoarDataTable.vue

```diff
@@ -365,81 +419,65 @@ const compressedRows = ref(false);
 const valid_dataTypes = ['NUMERIC', 'NUMBER', 'TEXT', 'STRING', 'DATE', 'BOOLEAN', 'SCORE', 'PROGRESS'];
-let filters = {};
-let options = {};
-_forEach(computedColumns.value, (column) => {
-  // Check if header text is supplied; if not, generate.
-  if (!_get(column, 'header')) {
-    column['header'] = _startCase(_get(column, 'field'));
-  }
-  const dataType = _toUpper(_get(column, 'dataType'));
-  let returnMatchMode = null;
-  if (valid_dataTypes.includes(dataType)) {
-    if (dataType === 'NUMERIC' || dataType === 'NUMBER' || dataType === 'BOOLEAN') {
-      returnMatchMode = { value: null, matchMode: FilterMatchMode.EQUALS };
-    } else if (dataType === 'TEXT' || dataType === 'STRING') {
-      returnMatchMode = { value: null, matchMode: FilterMatchMode.EQUALS };
-    } else if (dataType === 'DATE') {
-      returnMatchMode = { value: null, matchMode: FilterMatchMode.DATE_IS };
-    } else if (dataType === 'SCORE') {
-      // The FilterMatchMode does not matter as we are using this in conjunction with 'lazy',
-      //   so the filter event is being handled in an external handler.
-      if (scoredTasks.includes(column.field.split('.')[1])) {
-        returnMatchMode = { value: null, matchMode: FilterMatchMode.STARTS_WITH };
-      }
-    } else if (dataType === 'PROGRESS') {
-      returnMatchMode = { value: null, matchMode: FilterMatchMode.STARTS_WITH };
+const computedFilters = computed(() => {
+  let filters = {};
+  let options = {};
+  _forEach(computedColumns.value, (column) => {
+    // Check if header text is supplied; if not, generate.
+    if (!_get(column, 'header')) {
+      column['header'] = _startCase(_get(column, 'field'));
     }
+    const fieldOrFilterField = column?.filterField ? column.filterField : column.field;
+    const dataType = _toUpper(_get(column, 'dataType'));
+    let returnMatchMode = null;
+    if (valid_dataTypes.includes(dataType)) {
+      if (dataType === 'NUMERIC' || dataType === 'NUMBER' || dataType === 'BOOLEAN') {
+        returnMatchMode = { value: null, matchMode: FilterMatchMode.EQUALS };
+      } else if (dataType === 'TEXT' || dataType === 'STRING') {
+        returnMatchMode = { value: null, matchMode: FilterMatchMode.CONTAINS };
+      } else if (dataType === 'DATE') {
+        returnMatchMode = { value: null, matchMode: FilterMatchMode.DATE_IS };
+      } else if (dataType === 'SCORE') {
+        returnMatchMode = { value: null, matchMode: FilterMatchMode.CONTAINS };
+      } else if (dataType === 'PROGRESS') {
+        returnMatchMode = { value: null, matchMode: FilterMatchMode.CONTAINS };
+      }
 
-    if (_get(column, 'useMultiSelect')) {
-      returnMatchMode = { value: null, matchMode: FilterMatchMode.IN };
-      options[column.field] = getUniqueOptions(column);
+      if (_get(column, 'useMultiSelect')) {
+        returnMatchMode = { value: null, matchMode: FilterMatchMode.IN };
+        options[column.field] = getUniqueOptions(column);
+      }
     }
```

Comment:
I had a hard time formatting this as a suggestion due to deleted lines between code blocks, but I hope you get the idea! I am also assuming that useMultiSelect is a boolean or undefined.
```
const useMultiSelect = _get(column, 'useMultiSelect'. false)
const inMatchMode = { value: null, matchMode: FilterMatchMode.IN }

const returnMatchMode = useMultiSelect ? inMatchMode : getMatchMode(dataType)

if (useMultiSelect) {
    options[column.field] = getUniqueOptions(column)
}
```